atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClient.java

```diff
@@ -65,30 +73,84 @@ public CassandraClient(Cassandra.Client delegate, AtlasDbQosClient qosClient) {
             SlicePredicate predicate, ConsistencyLevel consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
         qosClient.checkLimit();
-        return delegate.multiget_slice(keys, column_parent, predicate, consistency_level);
+        Map<ByteBuffer, List<ColumnOrSuperColumn>> result = delegate.multiget_slice(keys, column_parent,
+                predicate, consistency_level);
+        try {
+            recordBytesRead(getApproximateReadByteCount(result));
+        } catch (Exception e) {
```

Comment:
As before I don't think there should be any need to catch exceptions from the metrics code so we can remove all of these.

Reply:
What if there is a NullPointerException or something wrong in our metrics calculation, we will end up failing Cassandra operations if we dont catch these. Ideally, we dont want to throw an exception if recording metrics throws, right.

Reply:
Let's move the try/catch into the `recordBytesRead` method